### DISCUSSION TITLE
Nested conditional timelines

### DISCUSSION THREAD

**Discussion by AnneVogt** [jspsych/jsPsych#1916] [Q&A] [answered]
I am planning a questionnaire at the end of one of my experiments and depending on which answers participants give they will receive further questions or not. To this end, I have a row of several conditional timelines. 
If participants choose "yes" in the first question they receive some other questions and ideally the experiment would end then. However, this does not work because the second conditional timeline is evaluated and I am struggling to pass the information that the first question was answered with a "yes" to this condition. I was thinking of saving this answer into some kind of normal experiment variable and calling it afterwards but I didn't succeed. Therefore, I would appreciate your help.

```
var yes_no_options = ["ja", "nein"];

var instrument_now = {
    type: 'survey-multi-choice',
    questions: [
      {prompt: "Spielen Sie aktuell aktiv ein Instrument?", name: 'instrument_now', options: yes_no_options, required:true, horizontal: true}, ]
};

var childhood_instrument = {
    type: 'survey-multi-choice',
    questions: [
      {prompt: "Haben Sie als Kind ein Instrument erlernt?", name: 'childhood_instrument', options: yes_no_options, required:true, horizontal: true}, ]
};
 
var if_node1 = {
    timeline: [which_instrument, age_instrument, education_instrument, niveau_instrument, frequency_instrument],
    conditional_function: function(){
        // get the data from the previous trial,
       var text_response = JSON.parse(jsPsych.data.getLastTrialData().values()[0].responses);
		console.log(text_response);
		if (text_response.instrument_now == "ja"){
            return true;
        } else {
            return false;
        }
    },
  };

var if_node2 = {
    timeline: [childhood_instrument],
    conditional_function: function(){
var text_response2 = JSON.parse(jsPsych.data.getLastTrialData().values()[0].responses);
		console.log(text_response2);
		if (text_response2.instrument_now== "ja"){
            return false;
        } else {
            return true;
        }
    }
};

timeline.push(instrument_now, if_node1, if_node2)
```

The last part can not be evaluated properly if the first conditional timeline has been executed because then the values from the last trial are not "yes" or "no" anymore. How can I pass the information from the variable "text_response" to the second conditional timeline?

**Comment by becky-gilbert** [comment, marked as answer]
Hi @AnneVogt! Instead of getting the previous responses using `getLastTrialData`, what I would do here is use jsPsych's [data `filter` function](https://www.jspsych.org/core_library/jspsych-data/#filter) so that you can pick out the trial data that you need, regardless of whether or not the conditionals have run. The easiest way to do this is to add some unique labels to your questions using the [trial's `data` parameter](https://www.jspsych.org/overview/plugins/#the-data-parameter) - this way you'll be able to filter on these labels. Here's an example:
```js
var instrument_now = {
    type: 'survey-multi-choice',
    questions: [
      {prompt: "Spielen Sie aktuell aktiv ein Instrument?", name: 'instrument_now', options: yes_no_options, required:true, horizontal: true}, ],
    data: {question_name: 'instrument_now'}  // add a label to the trial data
};

var childhood_instrument = {
    type: 'survey-multi-choice',
    questions: [
      {prompt: "Haben Sie als Kind ein Instrument erlernt?", name: 'childhood_instrument', options: yes_no_options, required:true, horizontal: true}, ],
    data: {question_name: 'childhood_instrument'} // add a label to the trial data
};
```
Then, whenever you want to get the participant's responses to these questions, you can filter the jsPsych data like this:
```js
jsPsych.data.get().filter({question_name: 'instrument_now'}).values()[0].responses
jsPsych.data.get().filter({question_name: 'childhood_instrument'}).values()[0].responses
```
Does this solution work for you?

**Comment by AnneVogt** [comment]
Thanks you Becky for your quick response. I was able to implement your solution and now the different questions are displayed in the way I intended.
